src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+        m_settings.setValue(m_BILL_FUNDING_STATUS_KEY, fundingStatusBooleanToString(false));
+        m_settings.endGroup();
+        m_settings.sync();
+
+        // Create a new Bill object for the bill information inputted into the BillWidget
+        Bill enteredBill;
+        enteredBill.setName(m_billWidget->getNameInput()->text());
+        enteredBill.setAmountDue(m_billWidget->getAmountDueInput()->text().toDouble());
+        enteredBill.setDueDate(m_billWidget->getDueDateInput()->date());
+
+        // Default the bill to not having been funded yet
+        enteredBill.setFundedStatus(false);
+
+        // Insert the Bill object into the map with a key of the name of the bill with spaces removed
+        m_billMap[removeSpaces(m_billWidget->getNameInput()->text())] = enteredBill;
+    }
+}
+
+void MainWindow::createMissingBillDetailsBox()
+{
+        // Create the missing bill name QMessageBox with appropriate displayed text
+        QMessageBox missingBillDetailsBox;
+        missingBillDetailsBox.setText(m_MISSING_BILL_DETAILS_BOX_PRIMARY_TEXT);
+        missingBillDetailsBox.setInformativeText(m_MISSING_BILL_DETAILS_BOX_INFO_TEXT);
+
+        // Give the user the options of "Ok'ing" the error
+        missingBillDetailsBox.setStandardButtons(QMessageBox::Ok);
+
+        // Store the user's selection
+        int missingBillDetailsBoxSelection = missingBillDetailsBox.exec();
+
+        switch(missingBillDetailsBoxSelection)
+        {
+            // If the user selected Ok
+            case QMessageBox::Ok :
+            {
+                // Have the user return to the bill widget to enter the missing bill name
+                break;
+            }
+
+            default:
+            {
+                // Default to have the user return to the bill widget to enter the missing bill name
+                break;
+            }
+        }
+}
+
+void MainWindow::saveBillAndDisplayBillWidget()
+{
+    // If the bill name or amount due fields are empty
+    if(m_billWidget->getNameInput()->text().isEmpty() || m_billWidget->getAmountDueInput()->text().isEmpty())
+    {
+        // Create the missing bill details message box
+        createMissingBillDetailsBox();
+    }
+
+
+    // Otherwise if the bill fields have been filled out appropriately
+    else
+    {
+        // Attempt to open the config file, if successful create a new Bill object and store contents in the config file
+        openConfigForBillCreation();
+
+        // Clear the BillWidget if the user elects to add another bill
+        clearBillWidget();
+    }
+}
+
+void MainWindow::clearBillWidget()
+{
+    // Clear the BillWidget for entry of another bill
+    m_billWidget->getNameInput()->clear();
+    m_billWidget->getAmountDueInput()->clear();
+    m_billWidget->getDueDateInput()->setDate(QDate::currentDate());
+}
+
+void MainWindow::saveBillAndDisplayDashboard()
```

Comment:
Change verbiage to bill table widget rather than dashboard across the application.

Reply:
All instances of "dashboard" removed.